app/javascript/src/warn-dsit-organisation-change.js

```diff
@@ -0,0 +1,40 @@
+/*
+  Show a standard JavaScript `confirm` modal with a warning message if a new
+  user is being created whose email address is not in the DSIT whitelist and
+  their organisation is being set to DSIT, or when an existing user is being
+  edited whose email address is not in the DIST whitelist and their
+  organisation is being set to DSIT from a different organisation.
+*/
+document.addEventListener("DOMContentLoaded", function() {
+  const VALID_DSIT_DOMAINS = ["dsit.gov.uk", "odamanagement.org"];
+  const DSIT_ORGANISATION_INDEX = 0;
+  const WARNING = "Are you sure you wish to add this user's primary organisation as DSIT?";
+
+  const userForm = document.querySelector("form.user-form");
+
+  if (!userForm) return;
+
+  const checkEmail = email => VALID_DSIT_DOMAINS.includes(email.split("@").pop());
+  const getEmail = () => document.querySelector("#user-email-field").value;
+  const getOrganisation = () => document.querySelector("#user_organisation_id").selectedIndex;
```

Comment:
Could this be `getOrganisationIndex`? And could these latter two – which are called once – be inlined to L25:26 below?

Reply:
I'm incorrect here, they're called twice.